src/Controller/Component/SecurityComponent.php

```diff
@@ -174,17 +180,22 @@ public function requireAuth($actions)
      *
      * @param \Cake\Controller\Controller $controller Instantiating controller
      * @param string $error Error method
+     * @param \Cake\Controller\Exception\SecurityException $exception Additional debug info describing the cause,
+     * debug mode only
      * @return mixed If specified, controller blackHoleCallback's response, or no return otherwise
      * @see \Cake\Controller\Component\SecurityComponent::$blackHoleCallback
      * @link http://book.cakephp.org/3.0/en/controllers/components/security.html#handling-blackhole-callbacks
      * @throws \Cake\Network\Exception\BadRequestException
      */
-    public function blackHole(Controller $controller, $error = '')
+    public function blackHole(Controller $controller, $error = '', SecurityException $exception = null)
     {
         if (!$this->_config['blackHoleCallback']) {
+            if (Configure::read('debug') && $exception !== null) {
+                throw $exception;
```

Comment:
What if we only hide the message when debug = false and always raise the SecurityException. That way custom exception loggers can get more information about the problems that happen in production